text/0000-associated-constants-on-ints.md

```diff
@@ -0,0 +1,269 @@
+[std::f32::consts::SQRT_2](https://doc.rust-lang.org/std/f32/consts/constant.SQRT_2.html)
+
+[std::f64::DIGITS](https://doc.rust-lang.org/std/f64/constant.DIGITS.html)
+
+[std::f64::EPSILON](https://doc.rust-lang.org/std/f64/constant.EPSILON.html)
+
+[std::f64::INFINITY](https://doc.rust-lang.org/std/f64/constant.INFINITY.html)
+
+[std::f64::MANTISSA_DIGITS](https://doc.rust-lang.org/std/f64/constant.MANTISSA_DIGITS.html)
+
+[std::f64::MAX](https://doc.rust-lang.org/std/f64/constant.MAX.html)
+
+[std::f64::MAX_10_EXP](https://doc.rust-lang.org/std/f64/constant.MAX_10_EXP.html)
+
+[std::f64::MAX_EXP](https://doc.rust-lang.org/std/f64/constant.MAX_EXP.html)
+
+[std::f64::MIN](https://doc.rust-lang.org/std/f64/constant.MIN.html)
+
+[std::f64::MIN_10_EXP](https://doc.rust-lang.org/std/f64/constant.MIN_10_EXP.html)
+
+[std::f64::MIN_EXP](https://doc.rust-lang.org/std/f64/constant.MIN_EXP.html)
+
+[std::f64::MIN_POSITIVE](https://doc.rust-lang.org/std/f64/constant.MIN_POSITIVE.html)
+
+[std::f64::NAN](https://doc.rust-lang.org/std/f64/constant.NAN.html)
+
+[std::f64::NEG_INFINITY](https://doc.rust-lang.org/std/f64/constant.NEG_INFINITY.html)
+
+[std::f64::RADIX](https://doc.rust-lang.org/std/f64/constant.RADIX.html)
+
+[std::f64::consts::E](https://doc.rust-lang.org/std/f64/consts/constant.E.html)
+
+[std::f64::consts::FRAC_1_PI](https://doc.rust-lang.org/std/f64/consts/constant.E.html)
+
+[std::f64::consts::FRAC_2_PI](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_2_PI.html)
+
+[std::f64::consts::FRAC_2_SQRT_PI](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_2_SQRT_PI.html)
+
+[std::f64::consts::FRAC_1_SQRT_2](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_1_SQRT_2.html)
+
+[std::f64::consts::FRAC_PI_2](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_2.html)
+
+[std::f64::consts::FRAC_PI_3](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_3.html)
+
+[std::f64::consts::FRAC_PI_4](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_4.html)
+
+[std::f64::consts::FRAC_PI_6](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_6.html)
+
+[std::f64::consts::FRAC_PI_8](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_8.html)
+
+[std::f64::consts::LN_2](https://doc.rust-lang.org/std/f64/consts/constant.LN_2.html)
+
+[std::f64::consts::LN_10](https://doc.rust-lang.org/std/f64/consts/constant.LN_10.html)
+
+[std::f64::consts::LOG2_E](https://doc.rust-lang.org/std/f64/consts/constant.LOG2_E.html)
+
+[std::f64::consts::LOG10_E](https://doc.rust-lang.org/std/f64/consts/constant.LOG10_E.html)
+
+[std::f64::consts::PI](https://doc.rust-lang.org/std/f64/consts/constant.PI.html)
+
+[std::f64::consts::SQRT_2](https://doc.rust-lang.org/std/f64/consts/constant.SQRT_2.html)
+
+Replace each item with an associated const value on the appropriate type. Deprecate the fourteen numeric-type modules and remove them from the documentation.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Deprecation warnings are annoying.
```

Comment:
There's an additional technical drawbacks to associated constants:
+ You are forged to reference them through the type; I agree with @Ekleog that it's less clear that we want to force this.
   - This technical limitation may be lifted by the language in the future, but for now it exists.